lib/Basics/FixedSizeAllocator.h

```diff
@@ -36,22 +36,24 @@ namespace arangodb {
 template <typename T>
 class FixedSizeAllocator {
  private:
+  // sizeof(T) is always a multiple of alignof(T) unless T is packed (which should never be the case here)!
+  static_assert((sizeof(T) % alignof(T)) == 0);
+
   class MemoryBlock {
    public:
+    void operator delete(void* p) {
+      ::operator delete(p);
+    }
+
     MemoryBlock(MemoryBlock const&) = delete;
     MemoryBlock& operator=(MemoryBlock const&) = delete;
 
     MemoryBlock(size_t numItems)
-        : _numAllocated(0), _numUsed(0), _alloc(nullptr), _data(nullptr) {
+        : _numAllocated(numItems), _numUsed(0), _next() {
       TRI_ASSERT(numItems >= 32);
-      // assumption is that the size of a cache line is at least 64,
-      // so we are allocating 64 bytes in addition
-      _alloc = new char[(std::max<size_t>(sizeof(T), alignof(T)) * numItems) + 64];
-
-      _numAllocated = numItems;
 
       // adjust memory address to cache line offset (assumed to be 64 bytes)
-      _data = reinterpret_cast<char*>((reinterpret_cast<uintptr_t>(_alloc) + 63u) &
+      _data = reinterpret_cast<char*>((reinterpret_cast<uintptr_t>(this + 1) + 63u) &
```

Comment:
Now we do have the cache-line math here, but also take into account the cache line overhead when we create the MemoryBlock with `operator new`. It would be ideal to deal with this just once.